src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArr = [];
+  let clonedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
```

Comment:
Consider will be better if you create a variables for these words and use them everywhere